x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/Watcher.java

```diff
@@ -221,6 +224,12 @@
     protected final boolean transportClient;
     protected final boolean enabled;
     protected final Environment env;
+    private SetOnce<EmailService> emailService = new SetOnce<>();
+    private SetOnce<HipChatService> hipChatService = new SetOnce<>();
```

Comment:
You can get rid of all the `SetOnce` instance definitions here again with the `List<>`, or?